packages/@aws-cdk/aws-gamelift/lib/game-server-group.ts

```diff
@@ -0,0 +1,563 @@
+}
+
+/**
+ * Creates a GameLift FleetIQ game server group for managing game hosting on a collection of Amazon EC2 instances for game hosting.
+ * This operation creates the game server group, creates an Auto Scaling group in your AWS account, and establishes a link between the two groups.
+ * You can view the status of your game server groups in the GameLift console.
+ * Game server group metrics and events are emitted to Amazon CloudWatch.
+ * Before creating a new game server group, you must have the following:
+ *  - An Amazon EC2 launch template that specifies how to launch Amazon EC2 instances with your game server build.
+ *  - An IAM role that extends limited access to your AWS account to allow GameLift FleetIQ to create and interact with the Auto Scaling group.
+ *
+ * To create a new game server group, specify a unique group name, IAM role and Amazon EC2 launch template, and provide a list of instance types that can be used in the group.
+ * You must also set initial maximum and minimum limits on the group's instance count.
+ * You can optionally set an Auto Scaling policy with target tracking based on a GameLift FleetIQ metric.
+ *
+ * Once the game server group and corresponding Auto Scaling group are created, you have full access to change the Auto Scaling group's configuration as needed.
+ * Several properties that are set when creating a game server group, including maximum/minimum size and auto-scaling policy settings, must be updated directly in the Auto Scaling group.
+ * Keep in mind that some Auto Scaling group properties are periodically updated by GameLift FleetIQ as part of its balancing activities to optimize for availability and cost.
+ *
+ * @see https://docs.aws.amazon.com/gamelift/latest/fleetiqguide/gsg-intro.html
+ *
+ * @resource AWS::GameLift::GameServerGroup
+ */
+export class GameServerGroup extends GameServerGroupBase {
+
+  /**
+   * Import an existing game server group from its attributes.
+   */
+  static fromGameServerGroupAttributes(scope: Construct, id: string, attrs: GameServerGroupAttributes): IGameServerGroup {
+    if (!attrs.gameServerGroupArn && !attrs.gameServerGroupName) {
+      throw new Error('Either gameServerGroupName or gameServerGroupArn must be provided in GameServerGroupAttributes');
+    }
+    const gameServerGroupName = attrs.gameServerGroupName ??
+      cdk.Stack.of(scope).splitArn(attrs.gameServerGroupArn!, cdk.ArnFormat.SLASH_RESOURCE_NAME).resourceName;
+
+    if (!gameServerGroupName) {
+      throw new Error(`No game server group name found in ARN: '${attrs.gameServerGroupArn}'`);
+    }
+
+    const gameServerGroupArn = attrs.gameServerGroupArn ?? cdk.Stack.of(scope).formatArn({
+      service: 'gamelift',
+      resource: 'gameservergroup',
+      resourceName: attrs.gameServerGroupName,
+      arnFormat: cdk.ArnFormat.SLASH_RESOURCE_NAME,
+    });
+
+    class Import extends GameServerGroupBase {
+      public readonly gameServerGroupName = gameServerGroupName!;
```

Comment:
Do we need the `!` here?

Reply:
Yes, as we are either extracting this value from ARN or getting this value from Attributes. and if value is null we are throwing an error but initial variable instance is String | undefined which create a compiler error